docs/migrate-from-sidecar-to-ambient/migration.adoc

```diff
@@ -0,0 +1,2114 @@
+if [[ "$AMBIENT_MODE" == "ambient" ]]; then
+  echo "PASS: Namespace in ambient mode: $AMBIENT_MODE"
+else
+  echo "FAIL: Namespace not in ambient mode: $AMBIENT_MODE"
+  exit 1
+fi
+
+# 2. Verify workloads are recognized by ztunnel
+WORKLOAD_COUNT=$(istioctl ztunnel-config workloads --namespace ztunnel | grep $NAMESPACE | grep HBONE | wc -l)
+if [[ "$WORKLOAD_COUNT" -gt 0 ]]; then
+  echo "PASS: Workloads detected by ztunnel: $WORKLOAD_COUNT"
+else
+  echo "FAIL: No workloads detected by ztunnel in $NAMESPACE"
+  exit 1
+fi
+
+# 3. Verify pods still have sidecars (should not be removed yet)
+SIDECAR_COUNT=$(kubectl get pods -n $NAMESPACE --no-headers | grep -c "2/2.*Running")
+if [[ "$SIDECAR_COUNT" -gt 0 ]]; then
+  echo "PASS: Sidecars still present: $SIDECAR_COUNT/$EXPECTED_PODS"
+else
+  echo "FAIL: Sidecars missing: $SIDECAR_COUNT/$EXPECTED_PODS"
+  exit 1
+fi
+
+# 4. Test connectivity still works
+TEST_POD=$(kubectl get pods -n $NAMESPACE -l app=reviews -o jsonpath='{.items[0].metadata.name}')
+HTTP_STATUS=$(kubectl exec $TEST_POD -n $NAMESPACE -- curl -s -w "%{http_code}" -o /dev/null http://reviews.$NAMESPACE:9080/reviews/1)
+if [[ "$HTTP_STATUS" == "200" ]]; then
+  echo "PASS: Service connectivity working in ambient mode"
+else
+  echo "FAIL: Service connectivity failed: HTTP $HTTP_STATUS"
+  exit 1
+fi
+
+# 5. Verify traffic is using HBONE protocol
+HBONE_TRAFFIC=$(istioctl ztunnel-config workloads --namespace ztunnel | grep $NAMESPACE | grep -c "HBONE")
+if [[ "$HBONE_TRAFFIC" -gt 0 ]]; then
+  echo "PASS: HBONE protocol active: $HBONE_TRAFFIC workloads"
+else
+  echo "FAIL: HBONE protocol not active"
+  exit 1
+fi
+
+echo "PASS: Step 6 validation passed - ready to proceed to Step 7"
+----
+
+**If validation fails**: Check ztunnel logs and verify namespace labels are correct.
+
+**Rollback Decision Point**: If Step 6 validation fails, disable ambient mode:
+[source,console]
+----
+# Disable ambient mode for the namespace
+kubectl label namespace bookinfo istio.io/dataplane-mode-
+# Traffic will continue through sidecars
+----
+**Risk**: High - This is the critical transition point. Test thoroughly before proceeding.
+
+[[step-7-enable-waypoint-processing]]
+=== Step 7: Enable Waypoint Processing
+
+[[step-71-activate-waypoints-for-l7-processing]]
+==== Step 7.1 Activate Waypoints for L7 Processing
+
+**Critical Sequencing**: Only activate waypoints AFTER all required namespaces are in ambient mode.
+
+[source,console]
+----
+# Enable waypoint for bookinfo first
+kubectl label namespace bookinfo istio.io/use-waypoint=waypoint
+
+# Verify waypoint activation
+kubectl get pods -n bookinfo -l gateway.networking.k8s.io/gateway-name=waypoint
+----
+
+[[step-72-validate-l7-policy-enforcement]]
+==== Step 7.2 Validate L7 Policy Enforcement
+
+Test that L7 policies are now enforced through waypoints (L7 policies should work as before after waypoint activation):
```

Comment:
By the time we are here, the bookinfo pods are still having the sidecar proxies. Why would the traffic go via the waypoints?